lib/matplotlib/axis.py

```diff
@@ -1434,6 +1434,13 @@ def update_units(self, data):
         if converter is None:
             return False
 
+        if (self.converter is not None) and (self.converter != converter):
+            raise TypeError('Attempting to plot data that is '
+                            'registered to be converted with %s but '
```

Comment:
Should the language be something like "data with units %s which is incompatible with existing axis units %s" - kinda wondering if registered convertors is too weedy for users.

Reply:
Open to this - I find the word "units" very confusing in this context.  Looking at the jpl stuff I can see there was an attempt to let the data have different units and then get converted by what the axis units were (i.e. if xunits='centimetres' but the x-axis units were 'meters' 1.0 would be plotted as 0.01) but that seems to have been lost along the way.  *Or* I'm misunderstanding (more likely)

Reply:
I think that's sort of what the basic units example might be about too-same numbers, but we'll do the conversion internally. Maybe just stick to type? "attempting to plot data of type %s that is incompatiable with existing axis data of type %s"

Reply:
Oooh, except it returns fun things like `matplotlib.dates.DateConverter` not the data type.  I could state what the new type is (i.e. type blah cannot be processed by axis unit converter `matplotlib.dates.DateConverter`)

Reply:
I like that last suggestions @jklymak 

FWIW a converter that converts meters should also be able convert centimeters fine (they're both lengths!)